### PR TITLE
[JENKINS-50598] - Add support of running JTH with a custom WAR

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I am experimenting with running PCT with WARs generated by Custom WAR Packager. I would like to be able to run JTH and PCT tests against WARs instead of the dependency-based mechanism. It would allow to support things like auto-configuration of WARs which is offered by WAR packager.

- [x] - Custom WAR can be passed via system property
- [x] - It is possible to prevent replacing of bundled plugins by older dependencies (opt-in, system property)
- [ ] - Some documentation

https://issues.jenkins-ci.org/browse/JENKINS-50598

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees @olivergondza

File: src/main/java/org/jvnet/hudson/test/TestPluginManager.java [hunk 1]
```diff
@@ -57,6 +66,11 @@
  * @see HudsonTestCase#useLocalPluginManager
  */
 public class TestPluginManager extends PluginManager {
+
+    @CheckForNull
+    private static final boolean DO_NOT_OVERRIDE_BUNDLED_PLUGINS_BY_OLDER_TEST_DEPS =
+            Boolean.getBoolean("jth.pluginManager.doNotOverrideBundledPluginsByOlderTestDeps");
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/TestPluginManager.java]: Maybe -0 on this: I feel it makes the semantics of functional tests a bit harder to understand, and for no very compelling reason: you can already simply ensure that the WAR under test contains sufficiently new versions of the dependencies.

**Comment by raul-arabaolaza** [comment]
@jglick I believe i have one valid use case, integration testing of shared libraries and Jenkinsfiles.

Now that CWP supports Configuration as code you can easily define a war file that ressembles a real environment, and with this you can use Jenkinsrule to perform the testing of your shared libraries using JTH with minimum hassle, you can even use the docker fixtures to set up auxiliary services like git repos and similar. 

WDYT?

**Comment by jglick** [comment]
> you can easily define a war file that ressembles a real environment, and with this you can use Jenkinsrule to perform the testing of your shared libraries using JTH

Not following. If you are using CWP, why would you not have the versions of plugins in CWP that you wanted?

**Comment by jglick** [comment]
@oleg-nenashev do you believe this is still still valid?

**Comment by oleg-nenashev** [comment]
Might be, but PCT already has a number of solutions applied for this issue.
Closing since I do not plan to work on this PR ATM

**Pull request closed without merging**